src/runtime/recipe/search.ts

```diff
@@ -11,7 +11,7 @@ export class Search {
   _phrase: string;
   _unresolvedTokens: string[];
   _resolvedTokens: string[];
-  constructor(phrase: string, unresolvedTokens = undefined) {
+  constructor(phrase: string, unresolvedTokens: string[] = undefined) {
```

Comment:
`string[] | undefined`

`[]` should* be different than `undefined`. 



`*` should, knowing well that JS identity is garbage.